docs/machineregistration-reference.md

```diff
@@ -104,6 +104,45 @@ The only required value for a successful installation is the `device` key as we
   ```
 </details>
 
+#### config.elemental.reset
+
+Contains the reset configuration that would be applied via `elemental-register --reset`, when booted from the recovery partition and passed to [`elemental reset`](https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental_reset.md)
+
+Supports the following values:
+
+| Key         | Type   | Default value | Description                                                                                                                                |
+|-------------|--------|---------------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| enabled           | bool   | false  | Allow reset on all registered machines  |
+| reset-persistent  | bool   | true   | Format the COS_PERSISTENT partition     |
```

Comment:
Format ?!

This probably shouldn't be boolean but rather have various values like "false", "format", "zero" (dd with /dev/zero), "random" (dd with /dev/random), and maybe others.

Reply:
Format as in `mkfs.ext4 -L COS_PERSISTENT /dev/vda5`, quite literally.

These flags are 1:1 matching the `elemental reset` cli arguments. 
The idea is that we may have an additional `shred-data` option and it should apply to all partitions marked for reset.

We didn't discuss having different (more granular) format settings per partition. It could be a nice requirement, but I think it will also need some arguments remapping on the elemental cli side.

Reply:
While `mkfs` fulfills the basic reset requirement, it still keeps an open security hole: The old data still exists.

It's ok to just offer a "format" option right now, but the final version must have "zero" and/or "random" options.